topic_dynamics/modeling.py

```diff
@@ -0,0 +1,343 @@
+    :return: Two matrices as DataFrames.
+    """
+    phi_matrix = model.get_phi().sort_index(axis=0)
+    phi_matrix.to_csv(os.path.abspath(os.path.join(output_dir, name + "_phi.csv")))
+    theta_matrix = model.get_theta().sort_index(axis=1)
+    theta_matrix.to_csv(os.path.abspath(os.path.join(output_dir, name + "_theta.csv")))
+
+
+@check_output_directory(output_dir="output_dir")
+def save_most_topical_files(theta_matrix: pd.DataFrame, tokens_file: str,
+                            n_files: int, output_dir: str, name: str) -> None:
+    """
+    Save the most topical files of the model.
+    When run several times, overwrites the data.
+    :param theta_matrix: Theta matrix.
+    :param tokens_file: the temporary file with tokens.
+    :param n_files: number of the most topical files to be saved for each topic.
+    :param output_dir: the output directory.
+    :param name: name of the processed dataset.
+    :return: None.
+    """
+    file2path = {}
+    with open(tokens_file) as fin:
+        for line in fin:
+            token_line = parse_token_line(line)
+            file2path[int(token_line.index)] = token_line.path
+    with open(os.path.abspath(os.path.join(output_dir, name + "_most_topical_files.txt")), "w+") as fout:
+        for i in range(1, theta_matrix.shape[0] + 1):
+            fout.write("Topic " + str(i) + "\n\n")
+            # Create a dictionary for this topic where keys are files and values are
+            # theta values for this topic and this file (only 10 largest)
+            topic_dict = theta_matrix.sort_values(by="topic_" + str(i), axis=1,
+                                                  ascending=False).loc["topic_" + str(i)][:n_files].to_dict()
+            for k in topic_dict.keys():
+                fout.write("{file_index};{topic_weight:.3f};{file_path}\n"
+                           .format(file_index=str(k), topic_weight=topic_dict[k], file_path=file2path[int(k)]))
+            fout.write("\n")
+
+
+def get_topics_weight(slices_file: str, theta_file: str) -> np.array:
+    """
+    Read the theta file and transform it into topic weights for different slices.
+    :param slices_file: the path to the file with the indices of the slices.
+    :param theta_file: the path tp the csv file with the theta matrix.
+    :return np.array of weights of each topic for each slice.
+    """
+    date2indices = {}
+    with open(slices_file) as fin:
+        for line in fin:
+            slice_line = parse_slice_line(line)
+            date2indices[slice_line.date] = (slice_line.start_index, slice_line.end_index)
+    topics_weight = []
+    with open(theta_file) as fin:
+        reader = csv.reader(fin)
+        next(reader, None)  # Skip the headers
+        for row in reader:
+            topics_weight.append([])
+            for date in date2indices.keys():
+                topics_weight[-1].append(sum(float(i) for i in row[date2indices[date][0]:date2indices[date][1] + 1]))
```

Comment:
max length of the line should be 99 characters

Reply:
please check it everywhere

Reply:
Redone it everywhere, I tried to make the breaks in lines reasonable, sometimes they are not all pretty, but automatic in PyCharm. Still tried to make them as pretty as I could.